### PR TITLE
api: add client-side weighted round robin LB policy config

### PR DESCRIPTION
**Pull request by markdroth**
Signed-off-by: Mark D. Roth <roth@google.com>

Commit Message: api: add client-side weighted round robin LB policy config
Additional Description: This is the config for a new LB policy that we are adding to gRPC.
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

CC @yousukseung @jaychenatr

### CODE REVIEW

File: api/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3/client_side_weighted_round_robin.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// This policy differs from the built-in ROUND_ROBIN policy in terms of
+// how the endpoint weights are determined. In the ROUND_ROBIN policy,
+// the endpoint weights are sent by the control plane via EDS. However,
+// in this policy, the endpoint weights are instead determined via
+// qps and CPU utilization metrics sent by the endpoint using the ORCA
+// protocol. The weight of a given endpoint is computed as
+// qps / cpu_utilization.
+//
+// See the :ref:`load balancing architecture overview<arch_overview_load_balancing_types>` for more information.
+message RoundRobin {
```
Comment (yousukseung) [inline api/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3/client_side_weighted_round_robin.proto:30]: Shouldn't it be ClientSideWeightedRoundRobin ?

Comment (markdroth) [reply api/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3/client_side_weighted_round_robin.proto]: Doh!  Good catch.  Fixed.

File: api/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3/client_side_weighted_round_robin.proto [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// [#protodoc-title: Client-Side Weighted Round Robin Load Balancing Policy]
+// [#not-implemented-hide:]
+
+// Configuration for the client_side_weighted_round_robin LB policy.
+//
+// This policy differs from the built-in ROUND_ROBIN policy in terms of
+// how the endpoint weights are determined. In the ROUND_ROBIN policy,
+// the endpoint weights are sent by the control plane via EDS. However,
+// in this policy, the endpoint weights are instead determined via
+// qps and CPU utilization metrics sent by the endpoint using the ORCA
```
Comment (suniltheta) [inline api/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3/client_side_weighted_round_robin.proto:25]: can you add full form of "ORCA"?

Comment (markdroth) [reply api/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3/client_side_weighted_round_robin.proto]: Done.

**Review by lizan — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**